From e766f6fbd54caa0f0b5bccb8af3386014c7549d0 Mon Sep 17 00:00:00 2001
From: shinichi-c <gokus12456@gmail.com>
Date: Tue, 25 Mar 2025 06:47:46 +0000
Subject: [PATCH 1/2] composer: bump android.hardware.graphics.common to v7

---
 interfaces/composer/Android.bp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/interfaces/composer/Android.bp b/interfaces/composer/Android.bp
index dcf78e9..11375e4 100644
--- a/interfaces/composer/Android.bp
+++ b/interfaces/composer/Android.bp
@@ -3,7 +3,7 @@ aidl_interface {
     system_ext_specific: true,
     srcs: [ "vendor/lindroid/composer/*.aidl" ],
     imports: [
-        "android.hardware.graphics.common-V5",
+        "android.hardware.graphics.common-V7",
     ],
 
     unstable: true,
-- 
2.52.0

