From 40b7468a11a7b9c1ddc8983d681243dbe77da0c8 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Thu, 28 Nov 2024 22:13:20 +0800
Subject: [PATCH] inputflinger: allow disabling input devices via idc

---
 services/inputflinger/reader/EventHub.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/services/inputflinger/reader/EventHub.cpp b/services/inputflinger/reader/EventHub.cpp
index c6054e7..7738087 100644
--- a/services/inputflinger/reader/EventHub.cpp
+++ b/services/inputflinger/reader/EventHub.cpp
@@ -2459,6 +2459,13 @@ void EventHub::openDeviceLocked(const std::string& devicePath) {
     // Obtain the associated device, if any.
     device->associatedDevice = obtainAssociatedDeviceLocked(devicePath, device->configuration);
 
+    // Disable device if device config property set
+    if (device->configuration &&
+        device->configuration->getBool("device.disabled")) {
+        device->disable();
+        ALOGV("Disabling device with id %d\n", device->id);
+    }
+
     // Figure out the kinds of events the device reports.
     device->readDeviceBitMask(EVIOCGBIT(EV_KEY, 0), device->keyBitmask);
     device->readDeviceBitMask(EVIOCGBIT(EV_ABS, 0), device->absBitmask);
-- 
2.52.0

