From 40622948e65f288ebb536f22d6d22d2b8442de23 Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Sat, 23 Aug 2025 12:23:03 +0800
Subject: [PATCH] Fix fingerprint when using app clone on mt6895 xaga

---
 libutils/binder/RefBase.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libutils/binder/RefBase.cpp b/libutils/binder/RefBase.cpp
index 7510aa2..5343956 100644
--- a/libutils/binder/RefBase.cpp
+++ b/libutils/binder/RefBase.cpp
@@ -493,6 +493,11 @@ void RefBase::incStrongRequireStrong(const void* id) const {
 void RefBase::decStrong(const void* id) const
 {
     weakref_impl* const refs = mRefs;
+    if (refs == nullptr) {
+        ALOGE("RefBase::decStrong called on object with null mRefs - possible use after free. "
+        "Object: %p, id: %p", this, id);
+        return;
+    }
     refs->removeStrongRef(id);
     const int32_t c = refs->mStrong.fetch_sub(1, std::memory_order_release);
 #if PRINT_REFS
-- 
2.52.0

