From 72bc2507fcdadbf1f01d946a4addc9c8d975be3c Mon Sep 17 00:00:00 2001
From: Hecheng Yu <kde-yyds@qq.com>
Date: Sat, 31 Jan 2026 23:21:39 +0800
Subject: [PATCH] Remove duplicate sepolicy

---
 basic/non_plat/genfs_contexts | 2 --
 1 file changed, 2 deletions(-)

diff --git a/basic/non_plat/genfs_contexts b/basic/non_plat/genfs_contexts
index 4cdce35..fa20d0f 100644
--- a/basic/non_plat/genfs_contexts
+++ b/basic/non_plat/genfs_contexts
@@ -44,8 +44,6 @@ genfscon proc /mtk_cmdq_debug/status         u:object_r:proc_cmdq_debug:s0
 genfscon proc /mtk_cmdq_debug/record         u:object_r:proc_cmdq_debug:s0
 genfscon proc /cpuhvfs/dbg_repo              u:object_r:proc_dbg_repo:s0
 genfscon proc /sys/kernel/panic_on_rcu_stall u:object_r:proc_panic_on_rcu_stall:s0
-genfscon proc /sys/vm/dirty_writeback_centisecs u:object_r:proc_dirty:s0
-genfscon proc /sys/kernel/sched_pelt_multiplier u:object_r:proc_sched:s0
 
 # Purpose dump not exit file
 genfscon proc /isp_p2/isp_p2_dump     u:object_r:proc_isp_p2_dump:s0
-- 
2.52.0

